src/components/form/fields/consentField.jsx

```diff
@@ -21,30 +24,38 @@ const Description = withTheme(styled.div`
     }
 `);
 
-const ConsentField = ({ value, onChange, label, description, path }) => {
+const ConsentField = ({ value, onChange, label, description, path, required, validation }) => {
     const clickUpdate = ({ value }) => ({
         value: !value,
         isDirty: true
     });
 
     return <div style={{ position: "relative" }}>
-        <Checkbox value={value}
+        <Checkbox
+            value={value}
             onToggle={() => onChange(clickUpdate)}
             name={path}
             label={label}
-            data-testid={path} />
+            {...pick(validation, 'error')}
+            data-testid={path}
+            required={required} />
         <MarkdownContent root={Description} source={description} />
     </div>
 };
 
 export default function consentField(config) {
-    return createField({
+    const baseProps = {
         ...config,
+        required: !!config.required,
         defaultValue: config.defaultValue && { granted: config.defaultValue },
         format: {
             bind: x => !!(x && x.granted),
             unbind: x => ({ granted: x, consentType: config.type })
         },
         component: ConsentField
-    });
+    }
+
+    const props = config.required ? { ...baseProps, validator: checked } : baseProps;
```

Comment:
You can keep the existing code and add `validator: config.required !! checked`

Reply:
This doesn't work because if the `required` property is `true`, the validator will be `true` and not  `checked`.

Reply:
My bad, you can use the `||` operator then, instead of the coalescence one.

Reply:
This doesn't work either, it will also return `true` instead of `checked` if the consent is required.

Reply:
I mean this logic has a better place inside the props itself. The implementation then doesn't matter.